src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/WindowBackdropManager.cs

```diff
@@ -0,0 +1,147 @@
+    }
+
+    private static bool ApplyBackdrop(IntPtr hwnd, WindowBackdropType backdropType)
+    {
+        UpdateGlassFrame(hwnd, backdropType);
+        
+        var backdropPvAttribute = backdropType switch
+        {
+            WindowBackdropType.Auto => Standard.DWMSBT.DWMSBT_TABBEDWINDOW,
+            WindowBackdropType.TabbedWindow => Standard.DWMSBT.DWMSBT_TABBEDWINDOW,
+            WindowBackdropType.MainWindow => Standard.DWMSBT.DWMSBT_MAINWINDOW,
+            WindowBackdropType.TransientWindow => Standard.DWMSBT.DWMSBT_TRANSIENTWINDOW,
+            _ => Standard.DWMSBT.DWMSBT_NONE
+        };
+
+        var dwmResult = NativeMethods.DwmSetWindowAttributeSystemBackdropType(hwnd, backdropPvAttribute);
+        return dwmResult == HRESULT.S_OK;
+    }
+
+    private static bool RemoveBackdrop(IntPtr hWnd)
+    {
+        UpdateGlassFrame(hWnd, WindowBackdropType.None);
+
+        var backdropPvAttribute = Standard.DWMSBT.DWMSBT_NONE;
+        var dwmResult = NativeMethods.DwmSetWindowAttributeSystemBackdropType(hWnd, backdropPvAttribute);
+        return dwmResult == HRESULT.S_OK;
+    }
+
+    private static bool RemoveBackground(IntPtr hwnd)
+    {
+        var windowSource = HwndSource.FromHwnd(hwnd);
+        if(windowSource?.Handle != IntPtr.Zero && windowSource.CompositionTarget != null)
+        {
+            windowSource.CompositionTarget.BackgroundColor = Colors.Transparent;
+            return true;
+        }
+        return false;
+    }
+
+    private static bool RestoreBackground(IntPtr hwnd)
+    {
+        var windowSource = HwndSource.FromHwnd(hwnd);
+        if(windowSource?.Handle != IntPtr.Zero && windowSource.CompositionTarget != null)
+        {
+            windowSource.CompositionTarget.BackgroundColor = SystemColors.WindowColor;
+            return true;
+        }
+        return false;
+    }
+
+    private static bool UpdateGlassFrame(IntPtr hWnd, WindowBackdropType backdropType)
+    {
+        MARGINS margins = new MARGINS();
+        if(backdropType != WindowBackdropType.None)
+        {
+            margins = new MARGINS { cxLeftWidth = -1, cxRightWidth = -1, cyTopHeight = -1, cyBottomHeight = -1 };                    
+        }
+
+        var dwmApiResult = NativeMethods.DwmExtendFrameIntoClientArea(hWnd, ref margins);
+        return new HRESULT((uint)dwmApiResult) == HRESULT.S_OK;
+    }
+
+    #endregion
+
+    #region Internal Properties
+
+    internal static bool IsBackdropEnabled
+    {
+        get
+        {
+            if (_isBackdropEnabled == null)
+            {
+                _isBackdropEnabled = true;
+                
+                if (FrameworkAppContextSwitches.DisableFluentWindowsThemeWindowBackdrop || !Utility.IsOSWindows11Insider1OrNewer || !ThemeManager.IsFluentWindowsThemeEnabled)
+                {
+                    _isBackdropEnabled = false;
+                }
+            }
+
+            return (bool)_isBackdropEnabled;
+        }
+    }
```

Comment:
btw. this could be

```C#
internal static bool IsBackdropEnabled => _isBackdropEnabled ??= !FrameworkAppContextSwitches.DisableFluentWindowsThemeWindowBackdrop && Utility.IsOSWindows11Insider1OrNewer && ThemeManager.IsFluentWindowsThemeEnabled 
```